urbs.py

```diff
@@ -927,17 +1085,59 @@ def split_columns(columns, sep='.'):
 
 def commodity_subset(com_tuples, type_name):
     """ Unique list of commodity names for given type. 
-    
+
     Args:
         com_tuples: a list of (site, commodity, commodity type) tuples
         type_name: a commodity type ('Stock', 'SupIm', 'Env' or 'Demand')
-        
+
     Returns:
         The set (unique elements) of commodity names of the desired type
     """
     return set(com for sit, com, com_type in com_tuples 
                if com_type == type_name)
 
+def commodity_subset_list(com_tuples, type_list):
```

Comment:
In this case, extending the original `commodity_subset` function to handle both a `type_name` (string) and `type_list` (list) argument. Example: http://stackoverflow.com/a/999278/2375855